Firestore/core/src/firebase/firestore/local/query_result.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_LOCAL_QUERY_RESULT_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_LOCAL_QUERY_RESULT_H_
+
+#include <utility>
+#include <vector>
+
+#include "Firestore/core/src/firebase/firestore/model/document_key_set.h"
+#include "Firestore/core/src/firebase/firestore/model/document_map.h"
+
+namespace firebase {
+namespace firestore {
+namespace local {
+
+/** The result of executing a query against the local store. */
+class QueryResult {
```

Comment:
Port of https://github.com/firebase/firebase-android-sdk/blob/2a2e789629a7dad13e1007b6cdec26c96e73defe/firebase-firestore/src/main/java/com/google/firebase/firestore/local/QueryResult.java#L23